include/pybind11/cast.h

```diff
@@ -889,14 +889,25 @@ template <typename type> class type_caster<std::reference_wrapper<type>> {
     operator std::reference_wrapper<type>() { return subcaster.operator subcaster_cast_op_type&(); }
 };
 
+/// Call class-specific delete if it exists or global otherwise. Can also be an overload set.
+template <typename T, typename = void_t<decltype(static_cast<void (*)(void *)>(T::operator delete))>>
+void call_operator_delete(T *p) { T::operator delete(p); }
+
+inline void call_operator_delete(void *p) { ::operator delete(p); }
```

Comment:
Whoops, that was for part of the change that I reverted, but forgot to move it back.  I'll put it back.